packages/payments-plugin/src/stripe/stripe.controller.ts

```diff
@@ -0,0 +1,137 @@
+import { BadRequestException, Controller, Headers, HttpStatus, Post, Req, Res } from '@nestjs/common';
+import {
+    ChannelService,
+    InternalServerError,
+    LanguageCode,
+    Logger,
+    Order,
+    OrderService,
+    PaymentMethod,
+    RequestContext,
+    TransactionalConnection,
+} from '@vendure/core';
+import { OrderStateTransitionError } from '@vendure/core/dist/common/error/generated-graphql-shop-errors';
```

Comment:
I had to use deep import here because the `OrderStateTransitionError` class exported by `@vendure/core` comes from the [Admin API generated types](https://github.com/vendure-ecommerce/vendure/blob/e227463a7f78bdb0fed51964ef2e7c9eb676a057/packages/core/src/common/index.ts#L6) whereas the error thrown by `OrderService.transitionToState` is a [Shop API generated type](https://github.com/vendure-ecommerce/vendure/blob/e227463a7f78bdb0fed51964ef2e7c9eb676a057/packages/core/src/service/services/order.service.ts#L68).

Reply:
Oh that's such a pain with those issues, I've run into that stuff myself. OK, this should be fine anyway - since it's in the monorepo, any refactors to core will trigger an error here that can be fixed at the same time.